### ISSUE TITLE
#gossipsecretavailable - The gossip secret for a given product is more broadly available than it needs to be

### ISSUE THREAD

**Issue by tinnightcap** [nubisproject/nubis-consul#86]
The gossip secret for a given product is available in the following locations

```
user_data of the consul server for a given product (not required)
CloudFormation parameters of the consul server for a given product (not required)
Unauthenticated HTTP interface of the Consul server (not required)
Filesystem of all hosts for a given product (This is a design requirement as the hosts need access to the gossip secret to join the gossip cluster to do service discovery)
```

This enables AWS users or IAM roles that can do any of these things to be able to obtain the secret

```
ec2:DescribeInstanceAttribute  http://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeInstanceAttribute.html
cloudformation:DescribeStacks http://docs.aws.amazon.com/AWSCloudFormation/latest/APIReference/API_DescribeStacks.html
ec2:RunInstances
```

Access to this secret enables someone who can spin up an instance in a product VPC to impersonate a service provider and get access to data intended for other valid instances

**Comment by tinnightcap** [comment]
This issue will be solved by:
https://github.com/Nubisproject/nubis-consul/issues/63

**Comment by gozer** [comment]
This has been resolved by #138 and related work
